libnavigation-core/src/main/java/com/mapbox/navigation/core/history/MapboxHistoryRecorder.kt

```diff
@@ -0,0 +1,41 @@
+package com.mapbox.navigation.core.history
+
+import com.mapbox.navigation.base.options.HistoryRecorderOptions
+import com.mapbox.navigation.core.MapboxNavigation
+
+/**
+ * Interface into retrieving mapbox navigation history files. Retrieve an instance
+ * of this interface through [MapboxNavigation.historyRecorder].
+ *
+ * @see [HistoryRecorderOptions] to enable the recorder.
+ */
+interface MapboxHistoryRecorder {
+    /**
+     * The file directory where the history files are stored.
+     * Use this to purge old history files, upload later, or
+     * to use them for replay.
+     *
+     * @see [HistoryRecorderOptions] to customize this location.
+     *
+     * @return absolute path to the directory with history files.
+     *     returns null when there is no file directory
+     */
+    fun fileDirectory(): String?
+
+    /**
+     * Store history and run callback when the operation has completed.
+     * @param result Callback which run when history has been saved
+     */
+    fun saveHistory(result: SaveHistoryCallback)
+}
+
+/**
+ * Callback which called as a result of [MapboxHistoryRecorder.saveHistory]
+ */
+fun interface SaveHistoryCallback {
+    /**
+     * @param filepath is null if [MapboxHistoryRecorder.saveHistory] called without any
+     * events received or the navigator has is not configured.
```

Comment:
> navigator has is not configured

Can this ever happen? We create the navigator when `MapboxNavigation` is created. Also, developers don't know what a `navigator` is, so this message is not helpful anyway.

Reply:
If there's a `historyDir` misconfiguration or if it's not accessible, I believe it also returns `null`. Could you confirm @vanmaxim?

Reply:
This is still relevant.

Reply:
Here are the docs on the native callback
```
/**
 * Callback which called as a result of `dumpHistory`
 *
 * @param `filepath` could be null if `dumpHistory` called without any actually received events
 * or if `HistoryRecorder` destroyed before `dumpHistory` ends it's work (as an asynchronous method).
 */
```

What's the suggestion here? I haven't had a much time to revise each comment so suggestions are welcome.